modules/gltf/editor/editor_scene_importer_blend.cpp

```diff
@@ -467,73 +454,55 @@ bool EditorFileSystemImportFormatSupportQueryBlend::query() {
 		EditorNode::get_singleton()->get_gui_base()->add_child(browse_dialog);
 	}
 
-	String path = EDITOR_GET("filesystem/import/blender/blender3_path");
+	String path = EDITOR_GET("filesystem/import/blender/blender_path");
 
 	if (path == "") {
 		// Autodetect
 		auto_detected_path = "";
+		Vector<String> find_paths;
 
 #if defined(MACOS_ENABLED)
-
 		{
-			Vector<String> mdfind_paths;
-			{
-				List<String> mdfind_args;
-				mdfind_args.push_back("kMDItemCFBundleIdentifier=org.blenderfoundation.blender");
-
-				String output;
-				Error err = OS::get_singleton()->execute("mdfind", mdfind_args, &output);
-				if (err == OK) {
-					mdfind_paths = output.split("\n");
-				}
-			}
-
-			bool found = false;
-			for (const String &found_path : mdfind_paths) {
-				found = _autodetect_path(found_path.path_join("Contents/MacOS"));
-				if (found) {
-					break;
+			List<String> mdfind_args;
+			mdfind_args.push_back("kMDItemCFBundleIdentifier=org.blenderfoundation.blender");
+
+			String output;
+			Error err = OS::get_singleton()->execute("mdfind", mdfind_args, &output);
+			if (err == OK) {
+				for (const String find_path : output.split("\n")) {
+					find_paths.push_back(find_path.path_join("Contents/MacOS/Blender"));
 				}
 			}
-			if (!found) {
-				found = _autodetect_path("/opt/homebrew/bin");
-			}
-			if (!found) {
-				found = _autodetect_path("/opt/local/bin");
-			}
-			if (!found) {
-				found = _autodetect_path("/usr/local/bin");
-			}
-			if (!found) {
-				found = _autodetect_path("/usr/local/opt");
-			}
-			if (!found) {
-				found = _autodetect_path("/Applications/Blender.app/Contents/MacOS");
-			}
 		}
+		find_paths.push_back("/opt/homebrew/bin/blender");
+		find_paths.push_back("/opt/local/bin/blender");
+		find_paths.push_back("/usr/local/bin/blender");
+		find_paths.push_back("/usr/local/opt/blender");
+		find_paths.push_back("/Applications/Blender.app/Contents/MacOS/Blender");
 #elif defined(WINDOWS_ENABLED)
 		{
 			char blender_opener_path[MAX_PATH];
 			DWORD path_len = MAX_PATH;
 			HRESULT res = AssocQueryString(0, ASSOCSTR_EXECUTABLE, ".blend", "open", blender_opener_path, &path_len);
-			if (res == S_OK && _autodetect_path(String(blender_opener_path).get_base_dir())) {
-				// Good.
-			} else if (_autodetect_path("C:\\Program Files\\Blender Foundation")) {
-				// Good.
-			} else {
-				_autodetect_path("C:\\Program Files (x86)\\Blender Foundation");
+			if (res == S_OK && ) {
```

Comment:
You removed a condition here
```suggestion
			if (res == S_OK) {
```
Unless you need some added condition